kernel/src/engine/arrow_expression/evaluate_expression.rs

```diff
@@ -177,15 +187,12 @@ fn evaluate_transform_expression(
         ));
     }
 
-    // Verify that the lengths match before attempting to zip them below.
-    if output_cols.len() != output_schema.fields_len() {
-        return Err(Error::generic(format!(
-            "Expression count ({}) doesn't match output schema field count ({})",
-            output_cols.len(),
-            output_schema.fields_len()
-        )));
+    // Verify we consumed all output schema fields
+    if output_schema_iter.next().is_some() {
+        return Err(Error::generic("Output schema mismatch"));
```

Comment:
Same comment as above but the other way round, I'd go for `Output schema mismatch - expected fewer columns`.